src/pybind/mgr/mds_autoscaler/module.py

```diff
@@ -0,0 +1,87 @@
+"""
+Automatically scale MDSs based on status of the file-system using the FSMap
+"""
+
+import errno
+import logging
+from mgr_module import MgrModule
+
+import orchestrator
+
+log = logging.getLogger(__name__)
+
+"""
+Some terminology is made up for the purposes of this module:
+
+ - "grow/shrink" - increase/decrease the MDSs in the system
+"""
+
+
+class MDSAutoscaler(MgrModule):
+    """
+    MDS autoscaler.
+    """
+    def __init__(self, *args, **kwargs):
+        super(MDSAutoscaler, self).__init__(*args, **kwargs)
+        self.fs_map = None
+
+    def create_mds(self, fs_name):
+        spec = orchestrator.StatelessServiceSpec(fs_name)
```

Comment:
Right now, we're doing some refactoring of the orchestrator interface. We're making heavy use of type annotations. Therefore, please add `mds_autoscaler/module.py` to the `[mypy]` section of `src/pybind/mgr/tox.ini`!

Reply:
PR updated with suggested changes